cpp/src/aztec/plonk/proof_system/prover/prover.cpp

```diff
@@ -665,15 +665,10 @@ template <typename settings> void ProverBase<settings>::add_plookup_memory_recor
 
     const fr eta = fr::serialize_from_buffer(transcript.get_challenge("eta").begin());
     const fr eta_sqr = eta.sqr();
-    // At this point in the algorithm, the w_1, w_2, w_3 polynomials have been copied into
-    // the wire_fft map, and inverse fourier transforms have been applied to the original polynomials.
-    // We need their lagrange-base forms, so we tap into the wire_fft map.
-    // fr* w_1 = key->polynomial_cache.get("w_1_fft").get_coefficients();
-    // fr* w_2 = key->polynomial_cache.get("w_2_fft").get_coefficients();
-    // fr* w_3 = key->polynomial_cache.get("w_3_fft").get_coefficients();
-    fr* w_1 = key->polynomial_cache.get("w_1_lagrange").get_coefficients();
-    fr* w_2 = key->polynomial_cache.get("w_2_lagrange").get_coefficients();
-    fr* w_3 = key->polynomial_cache.get("w_3_lagrange").get_coefficients();
+
+    std::span<fr> w_1 = key->polynomial_cache.get("w_1_lagrange");
+    std::span<fr> w_2 = key->polynomial_cache.get("w_2_lagrange");
+    std::span<fr> w_3 = key->polynomial_cache.get("w_3_lagrange");
     fr* w_4 = key->polynomial_cache.get("w_4_lagrange").get_coefficients();
```

Comment:
Why is this not a span

Reply:
I'm following the convention that we do not modify a span. Not entirely necessary as we discussed but better stylistically I think